container/state.go

```diff
@@ -209,6 +213,9 @@ func (s *State) Wait(ctx context.Context, condition WaitCondition) <-chan StateS
 
 	resultC := make(chan StateStatus, 1)
 
+	stopResult := s.waitStopResult
+	removeResult := s.waitRemoveResult
```

Comment:
Perhaps silly question; are there cases where we handle _both_  `waitStopResult` and `waitRemoveResult` ? Or are they effectively mutually exclusive (stop -> get result, remove -> get result); if that's the case we could have a single `waitResult` that's used by both, or would that make the code more complicated?

Reply:
The final result will be only one of them (whichever comes first). I think having only one result could also work. Will check.